packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -426,12 +426,12 @@ export abstract class BaseService extends Resource
       Annotations.of(this).addWarning('taskDefinition and launchType are blanked out when using external deployment controller.');
     }
 
-    this.serviceArn = this.getResourceArnAttribute(this.resource.ref, {
+    this.serviceName = this.getResourceNameAttribute(this.resource.attrName);
+    this.serviceArn = Stack.of(this).formatArn({
```

Comment:
Note: this *needs* to keep using the `getResourceArnAttribute()` helper - otherwise, things will break.

Reply:
Is there a better approach?  The `getResourceArnAttribute()` generates a token which can't be parsed for region, account, clusterName or serviceName.  Which means when creating the new service to get the ARN which can be leveraged to be imported you have to do this to generate the code.

Reply:
The problem here is most likely the `${cluster.clusterName}` part we are referencing here. We are investigating right now to confirm that 🙂.

Reply:
@madeline-k did you ever conclude your investigation about that suspicious `${cluster.clusterName}` expression causing problems?